src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,87 @@
+package duke.parser;
+
+import java.util.Arrays;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DefaultCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.exceptions.TaskException;
+
+
+/**
+ * Identifies command and executes its class respectively
+ */
+public class Parser {
+
+    /**
+     * Contains instructions to be entered by user to control system
+     */
+    public enum Instructions {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND
+    }
+
+    /**
+     * @param input user command is passed in as String
+     * @return command class of type Command
+     * @throws TaskException returns an exception under TaskException class
+     */
+    public Command parse(String input) throws TaskException {
+        String[] part = input.split(" ");
+        int index = 0;
+        String first_word = part[0].toUpperCase();
+
+        if (part[0].equals("mark") || part[0].equals("unmark") || part[0].equals("delete")) {
+            index = Integer.parseInt(part[1]) - 1;
+        }
+
+        if (!Arrays.stream(Instructions.values()).anyMatch(x -> (x.toString()).equals(first_word))) {
```

Comment:
Could you break down this conditional statement into smaller statements to avoid complicated code, so as to improve code readability?